lib/v8.js

```diff
@@ -247,37 +247,42 @@ Deserializer.prototype.readRawBytes = function readRawBytes(length) {
 
 function arrayBufferViewTypeToIndex(abView) {
   const type = ObjectPrototypeToString(abView);
-  if (type === '[object Int8Array]') return 0;
-  if (type === '[object Uint8Array]') return 1;
-  if (type === '[object Uint8ClampedArray]') return 2;
-  if (type === '[object Int16Array]') return 3;
-  if (type === '[object Uint16Array]') return 4;
-  if (type === '[object Int32Array]') return 5;
-  if (type === '[object Uint32Array]') return 6;
-  if (type === '[object Float32Array]') return 7;
-  if (type === '[object Float64Array]') return 8;
-  if (type === '[object DataView]') return 9;
-  // Index 10 is FastBuffer.
-  if (type === '[object BigInt64Array]') return 11;
-  if (type === '[object BigUint64Array]') return 12;
-  return -1;
+
+  switch (type) {
```

Comment:
How does this compare to an object or Map lookup?

Reply:
Object is slower compared to switch:

With switch:

```
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-debug benchmark/v8/serialize.js
v8/serialize.js n=1000000: 705,193.2134097881
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-debug benchmark/v8/deserialize.js
v8/deserialize.js n=1000000: 1,595,554.4661464228
```

With object:

```
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-v8-object benchmark/v8/serialize.js
v8/serialize.js n=1000000: 696,654.8116336825
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-v8-object benchmark/v8/deserialize.js
v8/deserialize.js n=1000000: 1,575,265.6004526608
```

Reply:
Map is faster on serialize but slower on deserialize:

With switch:

```
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-master benchmark/v8/serialize.js
v8/serialize.js n=1000000: 690,649.3698025825
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-master benchmark/v8/deserialize.js
v8/deserialize.js n=1000000: 1,582,109.8185933107
```

With Map:

```
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-v8-map benchmark/v8/serialize.js
v8/serialize.js n=1000000: 695,808.1758001995
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-v8-map benchmark/v8/serialize.js
v8/serialize.js n=1000000: 699,940.6800273677
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-v8-map benchmark/v8/deserialize.js
v8/deserialize.js n=1000000: 1,565,646.8890889874
➜  node git:(perf/v8-deserialize-serialize) ✗ ./node-v8-map benchmark/v8/deserialize.js
v8/deserialize.js n=1000000: 1,572,391.3241736297
```

Reply:
That doesn't make sense to me since the current serialize benchmark included in this PR doesn't even exercise this code path.

Reply:
It should be noted when running benchmarks you can't just run them once and compare because the results can vary. So you either need to run them manually a bunch of times to get a feel for the expected range or use the compare.js script to do all of this for you and tell you whether results are statistically significant.

Reply:
I changed the serialize benchmark to this:

```js
'use strict';

const common = require('../common.js');
const v8 = require('v8');

const bench = common.createBenchmark(main, {
  n: [1e6]
});

function main({ n }) {
  const abv = new BigUint64Array();
  bench.start();
  for (let i = 0; i < n; i++)
    v8.serialize({ a: 1, b: abv });
  bench.end(n);
}
```

Using that benchmark against a plain object, a plain null object, and a Map, all of them have either the same or less performance compared to the if/switch solution.